data/scripts/bin/create-virtual-device.sh.template

```diff
@@ -0,0 +1,14 @@
+# Create a sparse-raw file for the boot sector
+dd if=/dev/zero of=/tmp/boot.raw bs=1M seek=1 count=0
+
+# Configure a loop device for boot.raw
+losetup /dev/loop2 /tmp/boot.raw
+
+# Create a virtual device for passing to coreos-installer
+dmsetup create agent <<.
+0 2048 linear /dev/loop2 0
+{{.Partition0.StartSector}} {{.Partition0.Size}} linear /dev/sda1  0
+{{.Partition1.StartSector}} {{.Partition1.Size}} linear /dev/sda2  0
+{{.Partition2.StartSector}} {{.Partition2.Size}} linear /dev/sda3  0
+{{.Partition3.StartSector}} {{.Partition3.Size}} linear /dev/sda4  0
```

Comment:
Will these partition devices always be `/dev/sda...`? What if they are `/dev/nvme0n1p...`?

Reply:
Good point. Maybe we can extract the boot device first with something like:
```
fdisk -l | grep dev | grep boot | cut -f1 -d'1'
```

Reply:
If I understood correctly this script is running before ignition, right? If so maybe you can use `/dev/disk/by-id/coreos-boot-disk`. Should be a link to the actual boot device. See https://docs.fedoraproject.org/en-US/fedora-coreos/storage/#_referencing_block_devices_from_ignition.

Reply:
> /dev/disk/by-id/coreos-boot-disk

Actually this scripts runs after the ignition. I.e. we include the script as part of the ignition, but it runs only after the machine is completely booted.

Reply:
This also works:
```
lsblk -rfp | grep agentboot | cut -f1 -d '5'
```

Reply:
The maybe use `/dev/disk/by-partlabel/...`?

Reply:
> Then maybe use `/dev/disk/by-partlabel/...`?

You mean something like:
```
{{.Partition0.StartSector}} {{.Partition0.Size}} linear /dev/disk/by-partlabel/BIOS-BOOT  0
{{.Partition1.StartSector}} {{.Partition1.Size}} linear /dev/disk/by-partlabel/EFI-SYSTEM 0
{{.Partition2.StartSector}} {{.Partition2.Size}} linear /dev/disk/by-partlabel/boot  0
{{.Partition3.StartSector}} {{.Partition3.Size}} linear /dev/disk/by-partlabel/root 0
```

Should be fine. But is it unique? I mean if we have another disk with the same partition naming?

Reply:
Yes, I mean that.

It is not unique, you can potentially have multiple partitions with the same label, but that would be a broken system. I don't think that is an issue for our use case.